sdk/vaa/governance.go

```diff
@@ -23,3 +27,33 @@ func CreateGovernanceVAA(timestamp time.Time, nonce uint32, sequence uint64, gua
 
 	return vaa
 }
+
+// Compute the hash for cosmwasm contract instatiation params.
+// The hash is keccak256 hash(hash(hash(BigEndian(CodeID)), Label), Msg).
+// We compute the nested hash so there is no chance of bytes leaking between CodeID, Label, and Msg.
+func CreateInstatiateCosmwasmContractHash(codeId uint64, label string, msg []byte) [32]byte {
+	var expected_hash [32]byte
+
+	// hash(BigEndian(CodeID))
+	var codeId_hash [32]byte
+	keccak := sha3.NewLegacyKeccak256()
+	if err := binary.Write(keccak, binary.BigEndian, codeId); err != nil {
+		panic(fmt.Errorf("failed to write binary data: %d", codeId).Error())
```

Comment:
We should include the actual error in the panic:

```suggestion
		panic(fmt.Sprintf("failed to write binary data (%d): %v", codeId, err))
```